### PR TITLE
Session gone

### PR DESCRIPTION
**Pull request by notkriswagner**
Added logic to catch session gone error and ignore if this occurs during session termination. Otherwise the error is returned to the user.

### CODE REVIEW

File: tests/mock/test_mock_session_gone.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2017-2019 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#include <string.h>
+#include "../utils/test_setup.h"
+#include "../utils/mock_setup.h"
+
+void test_session_gone_during_query(void **unused) {
```
Comment (smtakeda) [inline tests/mock/test_mock_session_gone.c]: Don't ignore the error while running query.

File: tests/mock/test_mock_session_gone.c [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        ++counter;
+    }
+    if (status != SF_STATUS_EOF) {
+        dump_error(&(sfstmt->error));
+    }
+    assert_int_equal(status, SF_STATUS_EOF);
+    snowflake_stmt_term(sfstmt);
+
+    setup_mock_delete_connection_session_gone();
+    status = snowflake_term(sf);
```
Comment (smtakeda) [inline tests/mock/test_mock_session_gone.c]: This is the only case we want to ignore the session gone error.

File: tests/mock/test_mock_session_gone.c (hunk 1 above)
Comment (notkriswagner) [reply tests/mock/test_mock_session_gone.c]: Yea I don't ignore the error while running a query. The purpose of this test is to ensure that if this occurs when we run a query, then we return an error to the user. See below for when we check for an error.

File: tests/mock/test_mock_session_gone.c [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    assert_int_equal(status, SF_STATUS_SUCCESS);
+
+    /* query */
+    setup_mock_query_session_gone();
+    SF_STMT *sfstmt = snowflake_stmt(sf);
+    status = snowflake_query(sfstmt, "select 1;", 0);
+    if (status != SF_STATUS_ERROR_CONNECTION_NOT_EXIST) {
+        dump_error(&(sfstmt->error));
+    }
+    assert_int_equal(status, SF_STATUS_ERROR_CONNECTION_NOT_EXIST);
```
Comment (notkriswagner) [inline tests/mock/test_mock_session_gone.c]: This is where we check to make sure that an error is returned to the user if we get a session gone error code.

File: tests/mock/test_mock_session_gone.c (hunk 1 above)
Comment (smtakeda) [reply tests/mock/test_mock_session_gone.c]: oh, sorry about that. I took wrong.

**Pull request merged**